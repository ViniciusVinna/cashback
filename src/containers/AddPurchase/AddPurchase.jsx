import React, { forwardRef } from 'react';
import { FiPlus } from 'react-icons/fi';

import Ripple from 'components/Ripple';

import { AddPurchaseStyled } from './AddPurchase.styled';

const {
  AddPurchaseWidget,
  PurchaseButton,
  PurchaseButtonWrapper,
} = AddPurchaseStyled;

const AddPurchase = forwardRef((props, ref) => (
  <AddPurchaseWidget ref={ref} data-testid="add-purchase">
    <PurchaseButtonWrapper>
      <PurchaseButton>
        <FiPlus />
        <Ripple />
      </PurchaseButton>
    </PurchaseButtonWrapper>
    Cadastrar Compra
  </AddPurchaseWidget>
));

export default AddPurchase;
