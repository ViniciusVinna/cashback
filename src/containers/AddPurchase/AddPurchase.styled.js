import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const AddPurchaseStyled = {
  AddPurchaseWidget: styled('div')(({ theme }) => css`
    align-items: center;
    backface-visibility: hidden;
    background-color: ${theme.color.d100};
    border: 1px solid ${theme.color.n400};
    bottom: 0;
    box-shadow: ${theme.boxShadow.default};
    color: ${theme.color.b500};
    display: flex;
    font-size: ${theme.textSize.s};
    justify-content: center;
    padding: ${theme.space.xl} ${theme.space.s} ${theme.space.s};
    position: fixed;
    width: 100%;
    z-index: ${theme.zIndex.default};
  `),
  PurchaseButtonWrapper: styled('div')`
    align-items: center;
    left: 50%;
    position: absolute;
    top: 0;
    transform: translate(-50%, -50%);
  `,
  PurchaseButton: styled('button')(({ theme }) => css`
    align-items: center;
    background-color: ${theme.color.b500};
    border-radius: ${theme.borderRadius.m};
    border: none;
    box-shadow: ${theme.boxShadow.modal};
    display: flex;
    height: ${theme.iconSize.xxl};
    justify-content: center;
    outline: none;
    position: relative;
    width: ${theme.iconSize.xxl};

    & svg {
      font-size: ${theme.iconSize.s};
      color:${theme.color.n100};
    }
  `),
};
