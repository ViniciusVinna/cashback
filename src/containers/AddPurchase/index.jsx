import AddPurchase from './AddPurchase';

export default AddPurchase;
