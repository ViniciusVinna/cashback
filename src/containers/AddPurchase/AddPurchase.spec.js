import React from 'react';

import AddPurchase from './AddPurchase';

describe('AddPurchase', () => {
  it('it should render properly', () => {
    const { container } = render(
      <AddPurchase />
    );
    expect(container).toMatchSnapshot();
  });
});
