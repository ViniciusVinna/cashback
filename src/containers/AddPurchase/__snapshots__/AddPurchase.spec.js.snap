// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPurchase it should render properly 1`] = `
<div>
  <div
    class="css-5uhkd6"
    data-testid="add-purchase"
  >
    <div
      class="css-172jk2k"
    >
      <button
        class="css-rztmuv"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="5"
            y2="19"
          />
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
        </svg>
        <canvas
          height="0"
          style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          width="0"
        />
      </button>
    </div>
    Cadastrar Compra
  </div>
</div>
`;
